ambry-frontend/src/integration-test/java/com/github/ambry/frontend/S3IntegrationTest.java

```diff
@@ -0,0 +1,282 @@
+import static org.junit.Assert.*;
+
+
+/**
+ * Integration tests for Ambry S3 support.
+ */
+public class S3IntegrationTest extends FrontendIntegrationTestBase {
+  private static final MockClusterMap CLUSTER_MAP;
+  private static final VerifiableProperties FRONTEND_VERIFIABLE_PROPS;
+  private static final VerifiableProperties SSL_CLIENT_VERIFIABLE_PROPS;
+  private static final FrontendConfig FRONTEND_CONFIG;
+  private static final InMemAccountService ACCOUNT_SERVICE =
+      new InMemAccountServiceFactory(false, true).getAccountService();
+  private static final String DATA_CENTER_NAME = "localDc";
+  private static final String HOST_NAME = "localhost";
+  private static final String CLUSTER_NAME = "Cluster-name";
+  private static RestServer ambryRestServer = null;
+  private static NettyClient plaintextNettyClient = null;
+  private static NettyClient sslNettyClient = null;
+  private final ObjectMapper xmlMapper;
+
+  static {
+    try {
+      CLUSTER_MAP = new MockClusterMap();
+      File trustStoreFile = File.createTempFile("truststore", ".jks");
+      trustStoreFile.deleteOnExit();
+      FRONTEND_VERIFIABLE_PROPS = buildFrontendVProps(trustStoreFile);
+      SSL_CLIENT_VERIFIABLE_PROPS = TestSSLUtils.createSslProps("", SSLFactory.Mode.CLIENT, trustStoreFile, "client");
+      FRONTEND_CONFIG = new FrontendConfig(FRONTEND_VERIFIABLE_PROPS);
+      ACCOUNT_SERVICE.clear();
+      ACCOUNT_SERVICE.updateAccounts(Collections.singletonList(InMemAccountService.UNKNOWN_ACCOUNT));
+    } catch (Throwable t) {
+      throw new IllegalStateException(t);
+    }
+  }
+
+  /**
+   * Constructor for {@link S3IntegrationTest}.
+   */
+  public S3IntegrationTest() {
+    super(FRONTEND_CONFIG, sslNettyClient);
+    xmlMapper = new XmlMapper();
+  }
+
+  /**
+   * Sets up an Ambry frontend server.
+   * @throws Exception
+   */
+  @BeforeClass
+  public static void setup() throws Exception {
+    ambryRestServer = new RestServer(FRONTEND_VERIFIABLE_PROPS, CLUSTER_MAP, new LoggingNotificationSystem(),
+        SSLFactory.getNewInstance(new SSLConfig(FRONTEND_VERIFIABLE_PROPS)));
+    ambryRestServer.start();
+    plaintextNettyClient = new NettyClient("localhost", PLAINTEXT_SERVER_PORT, null);
+    sslNettyClient = new NettyClient("localhost", SSL_SERVER_PORT,
+        SSLFactory.getNewInstance(new SSLConfig(SSL_CLIENT_VERIFIABLE_PROPS)));
+  }
+
+  /**
+   * Shuts down the Ambry frontend server.
+   */
+  @AfterClass
+  public static void teardown() {
+    if (plaintextNettyClient != null) {
+      plaintextNettyClient.close();
+    }
+    if (sslNettyClient != null) {
+      sslNettyClient.close();
+    }
+    if (ambryRestServer != null) {
+      ambryRestServer.shutdown();
+    }
+  }
+
+  @Test
+  public void multipartUploadTest() throws Exception {
+    Account account = ACCOUNT_SERVICE.createAndAddRandomAccount();
+    for (Container container : account.getAllContainers()) {
+      doMultipartUploadTest(account, container, 2, 1024);
```

Comment:
Since the doMultipartUploadTest support partCount and partSize parameter, can we randomize different partCount and partSize here and run the test?

Reply:
Wei, will you randomize the partCount and partSize. I'll approve it but leave it to you.

Reply:
Sure, added randomization for partCount and partSize.